src/Stash/Driver/Ephemeral.php

```diff
@@ -113,6 +119,12 @@ public function storeData($key, $data, $expiration)
 
         $this->store[$this->getKeyIndex($key)] = array('data' => $data, 'expiration' => $expiration);
 
+        if ($this->memoryLimit > 0) {
+            while (count($this->store) && memory_get_usage() > $this->memoryLimit) {
+                array_shift($this->store);
+            }
+        }
+
```

Comment:
I've experimented with this in the past and this method can use up a surprising amount of CPU in situations where lots of tiny objects are stored. This is because `memory_get_usage` triggers garbage collection in order to return an accurate number, so every loop through would do this.

Instead I recommend evicting a percentage of the items in the cache (say, 50%) during each loop. This will have a bigger impact on memory usage seriously reducing the overhead.

Reply:
Makes sense.

I'll, probably, make this percents configurable, because of different use cases and cache contents.

Experimenting with this solution I've just found a memory leak when it used together with Memcached driver:  `Driver\Composite (Driver\Ephemeral, Driver\Memcache)`. It causes all items to be evicted from Ephemeral, then fatal out of memory error. Will investigate further.

Reply:
I've added another option `memoryLimitEvictionFactor`. My experiments show good results.